domain/agentprovisioner/state/state.go

```diff
@@ -0,0 +1,115 @@
+package state
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/canonical/sqlair"
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/controller"
+	"github.com/juju/juju/core/database"
+	"github.com/juju/juju/domain"
+	"github.com/juju/juju/environs/config"
+)
+
+type State struct {
+	*domain.StateBase
+}
+
+func NewState(factory database.TxnRunnerFactory) *State {
+	return &State{
+		StateBase: domain.NewStateBase(factory),
+	}
+}
+
+func (s *State) GetModelConfigKeyValues(
+	ctx context.Context,
+	keys []string,
+) (*config.Config, error) {
```

Comment:
This won't work @barrettj12, `Config` has a lot of opinions around what isn't and isn't valid config and you can't construct a Config without a bunch of other keys. It also tells the user that this is a full model config when it isn't. Until we can decouple some business logic type should only be used from the service layer call site and up.

Reply:
ok, then unfortunately I'll have to reimplement here some of the logic from the config methods

Reply:
done